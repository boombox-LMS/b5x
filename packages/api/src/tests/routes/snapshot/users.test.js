const supertest = require("supertest");

describe("Users routes should match expectations", () => {
  let app;
  let cookie = "";
  const apiPrefix = global.__SUPERTEST_API_PREFIX__;

  beforeAll(async () => {
    // Add a seeded test database to the app, available as app.get('db')
    app = await global.buildAppWithSeededTestDatabase({
      users: [{ persona: "completed" }],
    });
  });

  // Close the database connection, which allows Jest to exit gracefully
  afterAll(async () => {
    await app.get("db").destroy();
  });

  // users.profile --------------------------------------------------

  describe("users.profile matches the snapshot", () => {
    let responseBody;

    test("users.profile returns a response", async () => {
      await supertest(app)
        .get(apiPrefix + `users.profile?username=completed-1`)
        .set("Cookie", cookie)
        .then((res) => {
          responseBody = res.body;
        })
        .catch((e) => {
          throw e;
        });
    });

    test("users.profile matches the snapshot on file", () => {
      expect(responseBody).toMatchSnapshot();
    });
  });

  // users.modifyGroups ---------------------------------------------

  describe("users.modifyGroups matches the snapshot", () => {
    let responseBody;

    test("users.modifyGroups returns a response", async () => {
      const requestBody = {
        users: [
          {
            email: "blank-2@test.com",
            addToGroups: ["new-group-1", "new-group-2", "new-group-3"],
            removeFromGroups: [],
          },
          {
            email: "totally-new-user@test.com",
            addToGroups: ["new-group-1", "new-group-2"],
            removeFromGroups: [],
          },
          {
            email: "totally-new-user-2@test.com",
            addToGroups: [
              "new-group-1",
              "new-group-2",
              "new-group-3",
              "new-group-4",
            ],
            removeFromGroups: [],
          },
        ],
      };

      await supertest(app)
        .post(apiPrefix + "users.modifyGroups")
        .set("Cookie", cookie)
        .send(requestBody)
        .then((res) => {
          responseBody = res.body;
        })
        .catch((e) => {
          throw e;
        });
    });

    test("users.modifyGroups matches the snapshot", () => {
      expect(responseBody).toMatchSnapshot();
    });
  });
});
