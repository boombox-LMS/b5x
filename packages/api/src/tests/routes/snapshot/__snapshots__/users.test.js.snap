// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users routes should match expectations users.modifyGroups matches expectations users.modifyGroups matches the snapshot on file 1`] = `
[
  {
    "email": "blank-2@test.com",
    "groups": [
      "new-group-1",
      "new-group-2",
      "new-group-3",
    ],
    "id": 4,
  },
  {
    "email": "totally-new-user@test.com",
    "groups": [
      "new-group-1",
      "new-group-2",
    ],
    "id": 5,
  },
  {
    "email": "totally-new-user-2@test.com",
    "groups": [
      "new-group-1",
      "new-group-2",
      "new-group-3",
      "new-group-4",
    ],
    "id": 6,
  },
]
`;

exports[`Users routes should match expectations users.profile matches expectations users.profile matches the snapshot on file 1`] = `
{
  "badges": [
    {
      "description": "Bravely refused to be discouraged by failure.",
      "icon": "https://api.iconify.design/carbon/loop.svg",
      "title": "Persistence practitioner",
    },
  ],
  "email": "completed-1@test.com",
  "firstName": "Completed",
  "lastName": "User",
  "username": "completed-1",
}
`;
